---
title: RangeError
slug: Web/JavaScript/Reference/Global_Objects/RangeError
tags:
  - Error
  - JavaScript
  - Object
  - RangeError
translation_of: Web/JavaScript/Reference/Global_Objects/RangeError
---
<div>{{JSRef}}</div>

<p>El objeto <code><strong>RangeError</strong></code> indica un error cuando un valor no se encuentra dentro de un rango de valores permitidos.</p>

<h2 id="Sintaxis">Sintaxis</h2>

<pre class="syntaxbox"><code>new RangeError([<var>message</var>[, <var>fileName</var>[, <var>lineNumber</var>]]])</code></pre>

<h3 id="Parámetros">Parámetros</h3>

<dl>
 <dt><code>message</code></dt>
 <dd>Opcional. Mensaje de error</dd>
 <dt><code>fileName</code> {{non-standard_inline}}</dt>
 <dd>Opcional. Nombre del fichero que contiene el código donde se produjo el error</dd>
 <dt><code>lineNumber</code> {{non-standard_inline}}</dt>
 <dd>Opcional. Número de línea de código donde se produjo el error</dd>
</dl>

<h2 id="Descripción">Descripción</h2>

<p>Se lanza un <code>RangeError</code> cuando se pasa como parámetro a una función un número que no está dentro del rango que dicha función permite. Esto puede ocurrir cuando se intenta crear un array  con una longitud inválida con el constructor {{jsxref("Array")}}, o al pasar valores inválidos a los métodos numéricos {{jsxref("Number.toExponential()")}}, {{jsxref("Number.toFixed()")}} o {{jsxref("Number.toPrecision()")}}.</p>

<h2 id="Propiedades">Propiedades</h2>

<dl>
 <dt>{{jsxref("RangeError.prototype")}}</dt>
 <dd>Permite añadir propiedades a objetos <code>RangeError</code>.</dd>
</dl>

<h2 id="Métodos">Métodos</h2>

<p><code>RangeError</code> no contiene métodos, pero hereda a través de la <a href="https://developer.mozilla.org/es/docs/Web/JavaScript/Herencia_y_la_cadena_de_protipos">cadena de prototipos</a>.</p>

<h2 id="Instancias_de_RangeError">Instancias de <code>RangeError</code></h2>

<h3 id="Propiedades_2">Propiedades</h3>

<div>{{page('/es/docs/Web/JavaScript/Reference/Global_Objects/RangeError/prototype', 'Propiedades')}}</div>

<h3 id="Métodos_2">Métodos</h3>

<div>{{page('/es/docs/Web/JavaScript/Reference/Global_Objects/RangeError/prototype', 'Métodos')}}</div>

<h2 id="Ejemplos">Ejemplos</h2>

<h3 id="Utilizando_RangeError">Utilizando <code>RangeError</code></h3>

<pre class="brush: js">var check = function(num) {
  if (num &lt; MIN || num &gt; MAX) {
    throw new RangeError('El parámetro debe estar entre ' + MIN + ' y ' + MAX);
  }
};

try {
  check(500);
}
catch (e) {
  if (e instanceof RangeError) {
    // TODO: Manejar el error de rango
  }
}
</pre>

<h2 id="Especificaciones">Especificaciones</h2>

<table class="standard-table">
 <tbody>
  <tr>
   <th scope="col">Especificación</th>
   <th scope="col">Estado</th>
   <th scope="col">Comentario</th>
  </tr>
  <tr>
   <td>{{SpecName('ES3')}}</td>
   <td>{{Spec2('ES3')}}</td>
   <td>Definición inical.</td>
  </tr>
  <tr>
   <td>{{SpecName('ES5.1', '#sec-15.11.6.2', 'RangeError')}}</td>
   <td>{{Spec2('ES5.1')}}</td>
   <td> </td>
  </tr>
  <tr>
   <td>{{SpecName('ES6', '#sec-native-error-types-used-in-this-standard-rangeerror', 'RangeError')}}</td>
   <td>{{Spec2('ES6')}}</td>
   <td> </td>
  </tr>
 </tbody>
</table>

<h2 id="Compatibilidad_en_navegadores">Compatibilidad en navegadores</h2>

{{Compat("javascript.builtins.RangeError")}}

<h2 id="Vea_también">Vea también</h2>

<ul>
 <li>{{jsxref("Error")}}</li>
 <li>{{jsxref("RangeError.prototype")}}</li>
 <li>{{jsxref("Array")}}</li>
 <li>{{jsxref("Number.toExponential()")}}</li>
 <li>{{jsxref("Number.toFixed()")}}</li>
 <li>{{jsxref("Number.toPrecision()")}}</li>
</ul>
